---
title: "Tomo mouse NSC bulk RNAseq profiling"
subtitle: "DESeq2(Reproducible Analysis)"
author: "Ryan Miller"
date: '`r format(Sys.time(), "Last modified: %d %b %Y")`'
output:
  html_document:
    toc: yes
    toc_float: yes
code_folding: show
theme: journal
---

Files copied into the input folder on July 30, 2025
Original file locations

- **Sample sheet from nextflow round 2**  
  `/data/project/MillerLab/projects/mNSC/250730_mNSC_nf-core`
- **Salmon gene counts**  
  `/data/project/MillerLab/projects/mNSC/250730_mNSC_nf-core/sf`
- **Genome**  
  `/data/project/MillerLab/genomes/GRCm39_vM37`

Project Setup
This document runs a full differential expression analysis pipeline. For reproducibility, especially within a Docker container, first ensure all required packages are installed by running the scripts/00_setup.R script.

# R setup and variables
## libraries
```{r}
#| echo: true
#| message: false
library("tidyverse")
library("rmarkdown")
library("reactable")
```

## folders
```{r}
# Create output directories if they don't already exist
dir_output <- "./output/"
dir_graph <- "./graphs/"
dir.create(dir_output, showWarnings = FALSE, recursive = TRUE)
dir.create(dir_graph, showWarnings = FALSE, recursive = TRUE)
```

## metadata
```{r}
samples_file <- "./input/samples.csv" # Define the path to the samples file
samples <- read.csv(samples_file, header = TRUE, sep = ",")
samples <- remove_rownames(samples)
samples <- column_to_rownames(samples, var = "sample")

# Ensure factors are set with a specific reference level for modeling
samples$Host <- factor(samples$Host, levels = c("BL6", "NSG"))
samples$Driver <- factor(samples$Driver, levels = c("EGFRvIII", "EGFRvV", "PDGFRA"))

## Display as an interactive table
reactable(
  samples,
  searchable = TRUE,
  filterable = TRUE,
  resizable = TRUE
)

## summary
colnames(samples)
summarized_samples <- samples %>%
  group_by(Driver, Type, Host) %>%
  summarize(n = n(), .groups = "drop")

## Display the summarized data as an interactive table
reactable(
  summarized_samples,
  searchable = TRUE,
  filterable = TRUE,
  resizable = TRUE
)
```

## sample subsets
```{r}
# ns1_by_host_and_driver
subset_1 <- subset(samples, samples$Type=="NS1")
reactable(
  subset_1,
  searchable = TRUE,
  filterable = TRUE,
  resizable = TRUE
)

# bl6_by_type_and_driver
subset_2 <- subset(samples, samples$Host=="BL6")
reactable(
  subset_2,
  searchable = TRUE,
  filterable = TRUE,
  resizable = TRUE
)
```

## sf files
```{r}
# Assuming salmon files are in a subfolder named 'sf' inside 'input'
files_sf <- file.path("./sf", paste0(rownames(samples), ".sf"))
names(files_sf) <- rownames(samples)

# Check that all specified files exist
missing_files <- files_sf[!file.exists(files_sf)]
if (length(missing_files) > 0) {
  stop(
    "The following files are missing: ",
    paste(missing_files, collapse = ", ")
  )
} else {
  message("All sample files exist.")
}
```

# Project variables
## variables and checks
```{r}
# Extract column names and assign to variables
cn <- samples %>% colnames()
cn
vars <- setNames(as.list(cn), cn)

## Data checks
  var1 <- "Driver"
  var2 <- "Host"
  var3 <- "Type"
  var4 <- "SeqBatch"

if (exists("var1") &&
    exists("var2") &&
    exists("var3") &&
    exists("var4")) {
  vars <- c(var4, var3, var2, var1)
} else {
  stop("One or more variables not defined. Check input.")
}

# Ensure all variables in vars exist in the samples data frame
valid_vars <- as.character(vars[vars %in% colnames(samples)])
if (length(valid_vars) != length(vars)) {
  missing_vars <- setdiff(vars, colnames(samples))
  stop(paste(
    "The following in vars do not exist in samples df:",
    paste(missing_vars, collapse = ", ")
  ))
}

# add interaction variables
var_int_1 <- paste0(var1,":", var2)
var_int_2 <- paste0(var1, ":", var3)
var_int_3 <- paste0(var1, ":", var2, ":", var3)
vars_int <- c(var_int_1, var_int_2, var_int_3)

# complete variable for model
vars_2 <- c(vars, vars_int)
```

## GOI
```{r}
project_genes <- c("Egfr", "Pdgfra", "Pten", "Cdkn2a")
```

## graph annotations and colors
```{r}
anno_vars <- c(var1, var2, var3)
anno <- as.data.frame(samples[, anno_vars])
summarized_anno <- anno %>%
  group_by(across(all_of(anno_vars))) %>%
  summarize(n = n(), .groups = "drop")
# Display as an interactive table
reactable(
  summarized_anno,
  searchable = TRUE,
  filterable = TRUE,
  resizable = TRUE
)

# Get unique entries for each variable
unique_values_by_var <- lapply(anno, unique)
unique_values_by_var

anno_colors <- list(
  Host = c("NSG" = "gray", "BL6" = "black"),
  Driver = c(
    "EGFRvIII" = "blue",
    "EGFRvV" = "dodgerblue",
    "PDGFRA" = "red"
  ),
  Type = c(
    "NS1" = "lightgray",
    "NS2" = "darkgray",
    "GBO" = "black",
    "Tumor" = "purple"
  )
)
```

## DESeq2 model design
```{r}
# Validate that vars are properly defined and exist in samples df
if (!all(vars %in% colnames(samples))) {
  stop("One or more variables are not valid column names in samples.")
}

designs <- list(
  # Model 1: Simple model with only main effects.
  # Analyze the effect of Host or Driver averaged across other conditions.
  model_1 <- as.formula(paste("~", paste(vars, collapse = " + ")))
  
  # Model 2: Full model with an interaction term between Driver and Host and Type.
  # Use this to ask if the effect of the Host different for each Driver
  model_2 <- as.formula(paste("~", paste(
    var4, vars_int, collapse = " + "
  )))
)
print(designs)
```

# Analysis Pipeline
This section sources the external R scripts that perform the analysis. Each step builds upon the previous one.

# Source scripts to global env
## 1. inputs
```{r}
#| echo: true
#| message: false
source("./scripts/01_inputs.R")
```

## 2. data prep
```{r}
#| echo: true
#| message: false
source("./scripts/02_data_prep.R")
```

## 3. Quality control
```{r}
#| echo: true
#| message: false
source("./scripts/03_QC.R")
```

## 4. DESeq2
```{r}
#| echo: true
#| message: false
source("./scripts/04_DESeq.R")
```

# Session
```{r}
## R environment details, including loaded packages and their versions.
## This is useful for debugging and ensuring reproducibility.
sessionInfo()
```
