---
title: "mNSC DEG and EA"
author: "Ryan Miller"
format: html
editor: source
---

# Libraries

```{r results: 'hide'}
# hides the chunk output

library("styler")
library("tidyverse")
library("DESeq2")
library("rmarkdown")
```

# Imports

Comprehensive sample sheet from nextflow round 1 (270725)

Merge metadata and sample sheet

## salmon gene counts

Original file location - <sftp://cheaha.rc.uab.edu//data/project/MillerLab/projects/mNSC/250725_Tomo_mNSC/nf-core_rnaseq/star_salmon/salmon.merged.gene_counts.rds>

File copied into the *input* folder

```{r}
# Specify the path to your RDS file

file_path <- "/data/input/Data_250725/salmon.merged.gene_counts.rds"

# Load the RDS file into an R object
gene_counts <- readRDS(file_path)

# View the first few rows
head(gene_counts)

# Check the sample names
head(colnames(gene_counts))
head(colData(gene_counts))
```

## sample metadata

File copied into the *input* folder on July 30, 2025

```{r}
# Specify the path to original metadata

file_path <- "/data/input/220927_TomoSamples.csv"

# import metadata csv
metadata <- read.csv(file_path, header = TRUE, sep = ",")

metadata$id <- metadata$ID

metadata$id <- str_replace(metadata$id, "TK_", "")
metadata$id <- str_pad(metadata$id, width = 3, side = c("left"), pad = "0")
metadata$id <- paste0("TK", metadata$id)
str(metadata)

remove_rownames(metadata) %>% has_rownames()

# replace sf column with new location in project folder (/data/sf)

file_path <- "/data/sf/"
metadata$sf <- paste0(file_path, metadata$id, ".sf")

files_sf <- metadata$sf
```

## sample sheet

Files copied into the *input* folder on July 30, 2025

```{r}
# Specify the path to original metadata

file_path <- "/data/input/Data_250725/samples.csv"

# import metadata csv
samples <- read.csv(file_path, header = TRUE, sep = ",")

remove_rownames(samples) %>% has_rownames()

```

## merge sample sheets

```{r}

samples <- left_join(samples, metadata, join_by("id"=="id"))

samples <- column_to_rownames(samples, var = "id")

```

# dds data 

Files copied into the input folder on July 30, 2025 from 250730 nf-core:rna-seq star-salmon deseq2 QC

This rda file contains transcript-level data for two assays (counts and vst transformed counts)

```{r}
# Specify the path to your dds rda file
file_path <- "/data/input/Data_250730/deseq2.dds.Rdata" 
# Load the RDS file into an R object 
load(file_path, .GlobalEnv)

dds_qc <- dds
rm(dds)
# View the first few rows 
head(dds_qc) 
# Check the sample names
head(colnames(dds_qc))
head(colData(dds_qc))
# Check rownames
head(rownames(dds_qc))
```

## se import

```{r}
# Specify the path to your dds rds file
file_path <- "/data/input/Data_250730/null.merged.SummarizedExperiment.rds" 
# Load the transcript-level RDS file into an R object 
tx_se <- readRDS(file = file_path)
# View the first few rows 
head(tx_se) 
# Check the sample names
head(colnames(tx_se))
head(colData(tx_se))
```

# variables
## R script

```{r}
#| label: "variables script"
#| echo: true
#| file: variables.R

# Source the external R script
source(file.path("./scripts/variables.R")) 
```

## expt variables

```{r}
head(rowData(tx_se))
head(colData(tx_se))

var1 <- "Driver"
var2 <- "Host"
var3 <- "Type"
var4 <- "Clone"
var5 <- "Replicate"
var6 <- "LibraryBatch"
var7 <- "SeqBatch"

anno <- as.data.frame(samples[,c(var1, var2, var3)])

colnames(anno)
```

# dds
## R script

```{r}
#| label: "dds script"
#| echo: true
#| file: dds.R

# Source the external R script
source(file.path("./scripts/dds.R")) 
```
