---
title: "Tomo mouse NSC bulk RNAseq profiling"
subtitle: DESeq2
author: "Ryan Miller"
affiliation: "UAB"
date: '`r format(Sys.time(), "Last modified: %d %b %Y")`'
output:
  html_document:
    toc: yes
    toc_float: yes
editor_options:
  chunk_output_type: inline
  markdown: 
    wrap: sentence
---

Files copied into the input folder on July 30, 2025
Original file locations

- **Sample sheet from nextflow round 2**  
  `/data/project/MillerLab/projects/mNSC/250730_mNSC_nf-core`
- **Salmon gene counts**  
  `/data/project/MillerLab/projects/mNSC/250730_mNSC_nf-core/sf`
- **Genome**  
  `/data/project/MillerLab/genomes/GRCm39_vM37`

# R setup and variables
## libraries
```{r}
#| echo: true
#| message: false
library("tidyverse")
library("rmarkdown")
library("reactable")
```

## folders
```{r}
getwd()
dir_output <- "./output/"
dir.create(dir_output)
dir_graph <- "./graphs/"
dir.create(dir_graph)
getwd()
```

## metadata
```{r}
samples_file <- "./input/samples.csv" # Define the path to the samples file
samples <- read.csv(samples_file, header = TRUE, sep = ",")
samples <- remove_rownames(samples)
samples <- column_to_rownames(samples, var = "sample")

## Display as an interactive table
reactable(
  samples,
  searchable = TRUE,
  filterable = TRUE,
  resizable = TRUE
)

## summary
colnames(samples)
summarized_samples <- samples %>%
  group_by(Driver, Type, Host) %>%
  summarize(n = n(), .groups = "drop")

## Display the summarized data as an interactive table
reactable(
  summarized_samples,
  searchable = TRUE,
  filterable = TRUE,
  resizable = TRUE
)
```

## sample subsets
```{r}
# ns1_by_host_and_driver
subset_1 <- subset(samples, samples$Type=="NS1")
reactable(
  subset_1,
  searchable = TRUE,
  filterable = TRUE,
  resizable = TRUE
)
# bl6_by_type_and_driver
subset_2 <- subset(samples, samples$Host=="BL6")
reactable(
  subset_2,
  searchable = TRUE,
  filterable = TRUE,
  resizable = TRUE
)
```

## sf files
```{r}
samples$files_sf <- paste0("./sf/", rownames(samples), ".sf")
files_sf <- samples$files_sf
missing_files <- samples$files_sf[!file.exists(samples$files_sf)]
missing_files <- files_sf[!file.exists(files_sf)]
if (length(missing_files) > 0) {
  stop(
    "The following files are missing: ",
    paste(missing_files, collapse = ", ")
  )
} else {
  message("All files exist.")
}
all(file.exists(files_sf))
```

# Project variables
## variables and checks
```{r}
# Extract column names and assign to variables
cn <- samples %>% colnames()
cn
vars <- setNames(as.list(cn), cn)

## Data checks
## Ensure validity of column names
if (length(vars) >= 18) {
  var1 <- as.factor(vars[[6]])
  var2 <- as.factor(vars[[7]])
  var3 <- as.factor(vars[[10]])
  var4 <- as.factor(vars[[11]])
  var5 <- as.factor(vars[[18]])
} else {
  stop("The 'samples' df does not have enough columns. Please check input.")
}
if (exists("var1") &&
    exists("var2") &&
    exists("var3") &&
    exists("var5")) {
  var_list <- c(var1, var2, var3, var5)
} else {
  stop("One or more variables not defined. Check input.")
}

# Ensure all variables in var_list exist in the samples data frame
valid_vars <- as.character(var_list[var_list %in% colnames(samples)])
if (length(valid_vars) != length(var_list)) {
  missing_vars <- setdiff(var_list, colnames(samples))
  stop(paste(
    "The following in var_list do not exist in samples df:",
    paste(missing_vars, collapse = ", ")
  ))
}
```

## GOI
```{r}
project_genes <- c("Egfr", "Pdgfra", "Pten", "Cdkn2a")
```

## graph annotations and colors
```{r}
anno <- as.data.frame(samples[, valid_vars])
summarized_anno <- anno %>%
  group_by(across(all_of(valid_vars))) %>%
  summarize(n = n(), .groups = "drop")
# Display as an interactive table
reactable(
  summarized_anno,
  searchable = TRUE,
  filterable = TRUE,
  resizable = TRUE
)

# Get unique entries for each variable
unique_values_by_var <- lapply(anno, unique)
unique_values_by_var

anno_colors <- list(
  Host = c("NSG" = "gray", "BL6" = "black"),
  Driver = c(
    "EGFRvIII" = "blue",
    "EGFRvV" = "dodgerblue",
    "PDGFRA" = "red"
  ),
  Type = c(
    "NS1" = "lightgray",
    "NS2" = "darkgray",
    "GBO" = "black",
    "Tumor" = "purple"
  )
)
```

## design
```{r}
### Validate that vars are properly defined and exist in samples df
if (!all(c(var1, var2, var3, var5) %in% colnames(samples))) {
  stop("One or more variables are not valid column names in samples.")
}
des_design <- as.formula(paste("~", var5, "+", var3, "+", var1, "+", var2))
```

# Source scripts to global env
## inputs
```{r}
#| echo: true
#| message: false
#| local: false
#| results: 'markup'
source("./scripts/01_inputs.R")
knitr::read_chunk("./scripts/01_inputs.R")
```

## data prep
```{r}
#| echo: true
#| message: false
#| local: false
#| results: 'markup'
source("./scripts/02_data_prep.R")
```

## QC
```{r}
#| echo: true
#| message: false
#| local: false
#| results: 'markup'
source("./scripts/03_QC.R")
```

## DESeq2
```{r}
#| echo: true
#| message: false
#| local: false
#| results: 'markup'
source("./scripts/04_DESeq.R")
```

## Source scripts to HTML knit
```{r}
#| include: true
# This chunk reads code from child R scripts
# knitr::read_chunk("./scripts/01_inputs.R")
# knitr::read_chunk("./scripts/02_data_prep.R")
# knitr::read_chunk("./scripts/03_QC.R")
# knitr::read_chunk("./scripts/04_DESeq.R")
```

# Session
```{r}
## R environment details, including loaded packages and their versions.
## This is useful for debugging and ensuring reproducibility of the analysis.
sessionInfo()
```
