---
title: "mNSC DEG and EA"
author: "Ryan Miller"
format: html
editor: source
---

# Libraries

```{r results: 'hide'}
# hides the chunk output
library("styler")
library("tidyverse")
library("rmarkdown")
```

Imports
Sample sheet from nextflow round 2 /data/project/MillerLab/projects/mNSC/250730_mNSC_nf-core

salmon gene counts
Original file locations
/data/project/MillerLab/projects/mNSC/250730_mNSC_nf-core/sf

genome
/data/project/MillerLab/genomes/GRCm39_vM37

Files copied into the *input* folder on July 30, 2025

# Sample metadata
```{r}
# path to original metadata
samples_file <- "/data/input/samples.csv"

# import metadata csv
samples <- read.csv(samples_file, header = TRUE, sep = ",")
remove_rownames(samples) %>% has_rownames()


## samples to rownames
rownames(samples) <- NULL
samples <- column_to_rownames(samples, var = c("sample"))
```

## sf files
```{r}
samples$files_sf <- paste0("./sf/", rownames(samples), ".sf")
files_sf <- samples$files_sf
all(file.exists(files_sf))
```

# Variables
```{r}
#| label: "variables"
#| echo: true
#| file: variables.R

# Source the external R script
source(file.path("./scripts/01_variables.R")) 
```

## Other variables
```{r}
head(samples)
colnames(samples)

var1 <- "Host"
var2 <- "Driver"
var3 <- "Type"
var4 <- "Clone"
var5 <- "Replicate"
var6 <- "LibraryBatch"
var7 <- "SeqBatch"

anno <- as.data.frame(samples[,c(var1, var2, var3)])
colnames(anno)
```

# genesets
```{r}
#| label: "genesets script"
#| echo: true
#| file: variables.R

# Source the external R script
source(file.path("./scripts/01a_genesets.R")) 
```

# txi R script
```{r}
#| label: "txi script"
#| echo: true

# Source the external R script
source(file.path("./scripts/02_txi.R")) 
```

## txi checks
```{r}
counts_raw <- as.data.frame(txi$counts)
look_for <- c("Egfr", "Pdgfra", "Pten", "Cdkn2a")
counts_raw[rownames(counts_raw) %in% look_for, ]
```
# DDS
```{r}
#| label: "dds script"
#| echo: false

# Source the external R script
source(file.path("./scripts/03_dds.R"))

# Source the external R script
source(file.path("./scripts/04_vsd.R"))
```

## DDS QC
```{r}
#| label: "RNAseqQC script"
#| echo: true

# Source the external R script
source(file.path("./scripts/05_QC.R")) 
```

# Genes
```{r}
#| label: "gene filter"
#| echo: true

# Source the external R script
source(file.path("./scripts/06_genes_mouse.R")) 
```

# DDS filter
```{r}
#| label: "dds filter script"
#| echo: false

# Source the external R script
source(file.path("./scripts/07_dds_filt.R"))
```

# dds_filter QC
```{r}
#| label: "RNAseqQC filtered"
#| echo: true

# Source the external R script
source(file.path("./scripts/08_QC_filt.R")) 
```

# dds_pcg QC
```{r}
#| label: "RNAseqQC pcg"
#| echo: true

# Source the external R script
source(file.path("./scripts/09_QC_pcg.R")) 
```

# Session
```{r}
sessionInfo()
```